ui/src/dashboards/selectors/index.ts

```diff
@@ -15,6 +15,47 @@ export const getTimeRange = (state: AppState): TimeRange => {
   return state.ranges[contextID] || DEFAULT_TIME_RANGE
 }
 
+export const getTimeRangeWithTimezone = (state: AppState): TimeRange => {
+  const timeRange = getTimeRange(state)
+  const timeZone = getTimeZone(state)
+  // create a copy of the timeRange so as not to mutate the original timeRange
+  const newTimeRange = Object.assign({}, timeRange)
+  if (timeRange.type === 'custom' && timeZone === 'UTC') {
+    // check to see if the timeRange has an offset
+    newTimeRange.lower = setTimeToUTC(newTimeRange.lower)
+    newTimeRange.upper = setTimeToUTC(newTimeRange.upper)
+  }
+  return newTimeRange
+}
+
+export const setTimeToUTC = (date: string): string => {
```

Comment:
can you add a comment here and a link to this issue? just to give a little more context into what this function is used for?

Reply:
definitely